src/app/service/form.service.ts

```diff
@@ -0,0 +1,69 @@
+import { Injectable } from '@angular/core';
+import { EventTemplate } from '../entity/EventTemplate';
+import { FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { BetTemplate } from '../entity/BetTemplate';
+import { multipleChoiceValidator } from '../shared/validator/multiple-choice.validator';
+import { multipleChoiceOptionValidator } from '../shared/validator/multiple-choice-option.validator';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FormService {
+  constructor(private reactiveFormBuilder: FormBuilder) {}
+
+  prepopulateEventTemplateForm(
+    eventTemplate: EventTemplate,
+    formGroup: FormGroup,
+  ) {
+    formGroup.patchValue({
```

Comment:
You should try to avoid making functions that mutate their parameters, since it can lead to situations that are hard to debug.

Please remove this method and move the code to the class where the `formGroup` parameter is declared, so you can access it with `this.formGroup`